### PR TITLE
Sanitize `config.url`

### PR DESCRIPTION
**Pull request by JSAssassin**
addresses #58

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -87,6 +87,22 @@ function _createUrl(config) {
...
+function _sanitizeUrl(url) {
+  const urlParts = require('url').parse(url);
+  let user;
+  if(urlParts.auth) {
+    user = urlParts.auth.split(':')[0];
+  }
+  let sanitizedUrl;
+  if(user) {
+    sanitizedUrl =
+      `${urlParts.protocol}//${user}:****@${urlParts.host}${urlParts.path}`;
```
Comment (mattcollier) [inline lib/index.js:99]: my preference would be that we don't even log the user, @JSAssassin @aljones15 @davidlehn @dlongley @gannan08 thoughts?

Comment (mattcollier) [reply lib/index.js:99]: Which would also simply the implementation somewhat here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # bedrock-mongodb ChangeLog
 
+## 8.0.4 - 2020-11-05
```
Comment (mattcollier) [inline CHANGELOG.md:3]: date needs to be TBD or similar.

File: lib/index.js (hunk 1 above)
Comment (gannan08) [reply lib/index.js:99]: In a production deployment, I do not see the added value for logging the user.

Comment (JSAssassin) [reply lib/index.js:99]: @mattcollier @gannan08 agreed. I can update that.

File: lib/index.js [hunk 3]
```diff
@@ -87,6 +87,11 @@ function _createUrl(config) {
   return url;
 }
 
+function _sanitizeUrl(url) {
```
Comment (mattcollier) [inline lib/index.js:90]: Please move helper to bottom of the file along with any other helpers (if any).  Private functions like this are hoisted and available before they are defined.

Comment (JSAssassin) [reply lib/index.js]: Fixed - df652a8

File: lib/index.js [hunk 4]
```diff
@@ -745,3 +731,22 @@ function _dropCollections(callback) {
...
+  // this needs to come last
+  if(config.username) {
+    url +=
+      `?authSource=${config.connectOptions.authSource || 'admin'}`;
+  }
+  return url;
+}
+
+function _sanitizeUrl(url) {
+  const urlParts = require('url').parse(url);
```
Comment (mattcollier) [inline lib/index.js:750]: Looking forward to when this and every other lib is converted to ESM syntax.  Please do not add CommonJS require into functions like this.  Pull in `url` at the top of the file along with all other deps which will make it easier to convert later.

File: lib/index.js (hunk 1 above)
Comment (davidlehn) [reply lib/index.js]: NODE_ENV values could also be used to control this if it turns out more debug info would be useful.

Comment (mattcollier) [reply lib/index.js]: -1 to bringing NODE_ENV into this, at least at this time.

File: lib/index.js (hunk 4 above)
Comment (JSAssassin) [reply lib/index.js]: Updated - 08b9847

**Review by mattcollier — APPROVED** [review]

**Review by davidlehn — APPROVED** [review]

**Review by davidlehn — CHANGES_REQUESTED** [review]
I think there are a few more places to change.
- The errors also include config.url.  If those trigger, logged error messages would have full URL.
- _connect is logging url twice.  At the top and in the MongoClient.connect callback. (same message, which seems confusing)

**Review by davidlehn — APPROVED** [review]

**Pull request merged**